src/suppressor.c

```diff
@@ -27,6 +27,9 @@ static int suppressor_process_key(char *key);
 
 void suppressor_init(unsigned int new_flags)
 {
+	if (old_process_key)
+		return;
```

Comment:
Actually, this may be wrong. We do call `suppressor_init` a second time on purpose when transitioning from wordlist to incremental mode. I guess that worked fine because `crk_process_key` got reset to incremental mode's before that second call. But now the above check probably breaks it.

Reply:
I didn't verify but it sure looks like it. Did you run it until the "Disabling duplicate candidate password suppressor" appeared? After that, next `crk_process_key()` should go boom.

Reply:
Maybe we could just do this
```diff
diff --git a/src/suppressor.c b/src/suppressor.c
index d4537f19e..3b36476c9 100644
--- a/src/suppressor.c
+++ b/src/suppressor.c
@@ -66,7 +66,8 @@ void suppressor_init(unsigned int new_flags)
        flags = new_flags;
        status.suppressor_end = 0;
        status.suppressor_end_time = 0;
-       old_process_key = crk_process_key;
+       if (!old_process_key)
+               old_process_key = crk_process_key;
        crk_process_key = suppressor_process_key;
 }

```

Reply:
Hmm no that might not be right either, as `crk_process_key apparently changes over time.

Reply:
Perhaps this
```diff
diff --git a/src/suppressor.c b/src/suppressor.c
index 7fa3fef0d..bed090f54 100644
--- a/src/suppressor.c
+++ b/src/suppressor.c
@@ -27,6 +27,9 @@ static int suppressor_process_key(char *key);
 
 void suppressor_init(unsigned int new_flags)
 {
+       if (old_process_key == suppressor_process_key)
+               return;
+
        if (!flags) {
                if (!(new_flags & SUPPRESSOR_UPDATE))
                        return;
```